src/org/javarosa/core/services/transport/payload/ByteArrayPayload.java

```diff
@@ -104,34 +97,31 @@ public void writeExternal(DataOutputStream out) throws IOException {
         ExtUtil.writeString(out, ExtUtil.emptyIfNull(id));
     }
 
-    /*
-     * (non-Javadoc)
-     * @see org.javarosa.core.services.transport.IDataPayload#accept(org.javarosa.core.services.transport.IDataPayloadVisitor)
-     */
+    @Override
     public <T> T accept(IDataPayloadVisitor<T> visitor) {
         return visitor.visit(this);
     }
 
-    /*
-     * (non-Javadoc)
-     * @see org.javarosa.core.services.transport.IDataPayload#getPayloadId()
-     */
+    public byte[] getPayloadBytes() {
+        return payload;
```

Comment:
I realized that this passes back a byte array, which could be modified. Sorry.
What do you think of replacing this method with a `toString` method that returns the String that you need, instead of making the string in the caller?

Reply:
Yes, I could do that. Thanks for the idea.

Reply:
This method can now be removed, right? It wasn't part of the original API and isn't used anymore.

Reply:
It should be removed